edge-modules/api-proxy-module/templates/nginx_default_config.conf

```diff
@@ -38,25 +38,42 @@ http {
             rewrite ^(.*)$ /storage$1 last;
         }
         #endif_tag ${BLOB_UPLOAD_ROUTE_ADDRESS} 
-        #if_tag !${BLOB_UPLOAD_ROUTE_ADDRESS}
+        #if_tag boolean_expression[!(${BLOB_UPLOAD_ROUTE_ADDRESS})]
         if ($http_x_ms_version)
         {
             rewrite ^(.*)$ /parent$1 last;
         }
-        #endif_tag ${BLOB_UPLOAD_ROUTE_ADDRESS}          
+        #endif_tag boolean_expression[!(${BLOB_UPLOAD_ROUTE_ADDRESS})]          
 
         #if_tag ${DOCKER_REQUEST_ROUTE_ADDRESS}
         if ($uri ~ ^/v2*.)
         {
             rewrite ^(.*)$ /registry$1 last;
         }
         #endif_tag ${DOCKER_REQUEST_ROUTE_ADDRESS}
-        #if_tag !${DOCKER_REQUEST_ROUTE_ADDRESS}
+        #if_tag ${CONNECTED_ACR_ROUTE_ADDRESS}
+        location /v2 {
+            resolver 127.0.0.11;
+            proxy_http_version 1.1;
+            proxy_pass         http://${CONNECTED_ACR_ROUTE_ADDRESS};
```

Comment:
CONNECTED_ACR allows image push. Nginx default settings only allows max 1M request body. So if the image layer is larger than 1M, the image push will fail. The fix is adding a setting `client_max_body_size 1000G`;

Reply:
fixed!